src/test/lib/operators/operator_performance_data_test.cpp

```diff
@@ -116,13 +122,63 @@ TEST_F(OperatorPerformanceDataTest, TableScanPerformanceData) {
     table_wrapper->execute();
 
     const auto table_scan = std::make_shared<TableScan>(
-        table_wrapper, between_inclusive_(get_column_expression(table_wrapper, ColumnID{0}), 1, 2));
+        table_wrapper, between_inclusive_(get_column_expression(table_wrapper, ColumnID{0}), 1, 4));
     table_scan->execute();
 
     const auto& performance_data = dynamic_cast<TableScan::PerformanceData&>(*table_scan->performance_data);
+    /*
+     * TODO(anyone) actually, there could be 3 chunks with all rows matching, and none with binary search.
+     * However, the all-match shortcurt of the sorted segment search currently always assumes an exclusive between.
```

Comment:
Assumes sounds kind of problematic, doesn't it? Can this break with other betweens?

Reply:
There are two tests ` // Between scan`. Maybe change second comment to ` // Between scan on nullable columns`.

Reply:
> Assumes sounds kind of problematic, doesn't it? Can this break with other betweens?
Not really. The worst thing that can happen is that we perform binary searches where an all-match-shortcut would work too, i.e., some unnecessary work. But i'd expect the performance impact of this to be rather small.
The same is true for the no-match-shortcut (an inclusive between is assumed, which may cause unnecessary scans, but should not break anything)
https://github.com/hyrise/hyrise/blob/e3bb51cb6b93503eb3dcca5411a36344b2b226f9/src/lib/operators/table_scan/sorted_segment_search.hpp#L156